fastats/core/ast_transforms/transformer.py

```diff
@@ -54,30 +57,38 @@ def visit_Call(self, node):
             # function globals, such as `range`
             return node
         elif name in self._params:
+            new_name = self.new_name_from_call_name(name)
+            new_func = self._params[name]
             self._replaced[name] = self._globals[name]
             self._globals[name] = convert_to_jit(self._globals[name])
-            node = ast.copy_location(self.modify_function_name(node), node)
+            self._globals[new_name] = convert_to_jit(new_func)
+
+            new_node = ast.Call(
+                func=ast.Name(id=new_name, ctx=ast.Load()),
+                args=node.args, keywords=[]
+            )
+            ast.copy_location(new_node, node)
+            ast.fix_missing_locations(new_node)
+            return new_node
         else:
             # Lazy import because it's circular.
             from fastats.core.ast_transforms.processor import AstProcessor
-
             orig_inner_func = self._globals[node.func.id]
-            self._replaced[node.func.id] = orig_inner_func
-            proc = AstProcessor(
-                orig_inner_func, self._params,
-                self._replaced, self._new_funcs
-            )
-            new_inner_func = proc.process()
-            self._globals[node.func.id] = convert_to_jit(new_inner_func)
+
+            not_ufunc = not isinstance(orig_inner_func, np.ufunc)
+            not_builtin = not isbuiltin(orig_inner_func)
+            if not_ufunc and not_builtin:
```

Comment:
I think we should refactor out these checks to some common util function. See changes in #18 : https://github.com/pawroman/fastats/blob/7169f578892f9a72c2c14baa9bfd1ce2b7f9b9ec/fastats/core/ast_transforms/convert_to_jit.py#L15-L19

Reply:
Yep, agreed. I suggest we ignore it here and take the refactor from your PR.